distributed/client.py

```diff
@@ -2656,9 +2668,8 @@ def __aiter__(self):
         return self
 
     def __next__(self):
-        with self.lock:
-            if not self.futures and self.queue.empty():
-                raise StopIteration()
+        if self.is_empty():
```

Comment:
May also be able to replace the similar check in `__anext__`. That one doesn't use a lock though, and I'm not sure if that was intentional or not.